sentry-android-core/src/main/java/io/sentry/android/core/ActivityLifecycleIntegration.java

```diff
@@ -146,6 +147,7 @@ private void stopPreviousTransactions() {
   }
 
   private void startTracing(final @NotNull Activity activity) {
+    WeakReference<Activity> weakActivity = new WeakReference<>(activity);
```

Comment:
Please try to reproduce the issue with and without the fix using LeakCanary.
Upgrade leakcanary to the latest version -> https://github.com/square/leakcanary/releases/tag/v2.9.1

Reply:
Tried with `2.9.1` but it didn't find anything regardless of `WeakReference` or old implementation.